### PR TITLE
EVG-14908: add middleware to check pod secret

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-14908

### Description 
* Add secret to pod data model, which will be generated when the pod is created in the container service.
* Add middleware to check pod ID and secret are valid when a pod contacts the app server.

### Testing 
Standard tests for the DB connector and middleware.

### CODE REVIEW

File: rest/data/pod.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	if p == nil {
+		return errors.New("pod does not exist")
+	}
+	if secret != p.Secret {
+		return errors.New("pod secrets do not match")
+	}
+	return nil
+}
+
+type MockPodConnector struct {
```
Comment (julianedwards) [inline rest/data/pod.go:28]: Documentation

File: rest/route/middleware.go [hunk 2]
```diff
@@ -316,6 +316,46 @@ func (m *hostAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request,
 	next(rw, r)
 }
 
+type podAuthMiddleware struct {
+	sc data.Connector
+}
+
+func NewPodAuthMiddleware(sc data.Connector) gimlet.Middleware {
```
Comment (julianedwards) [inline rest/route/middleware.go:323]: Documentation.

File: rest/data/pod.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+func (c *DBPodConnector) CheckPodSecret(id, secret string) error {
+	p, err := pod.FindOneByID(id)
+	if err != nil {
+		return err
+	}
+	if p == nil {
+		return errors.New("pod does not exist")
+	}
+	if secret != p.Secret {
+		return errors.New("pod secrets do not match")
```
Comment (julianedwards) [inline rest/data/pod.go:23]: Nit: can this say something like `incorrect pod secret`?

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: rest/data/pod.go (hunk 3 above)
Comment (Kimchelly) [reply rest/data/pod.go]: Done.

File: rest/data/pod.go (hunk 1 above)
Comment (Kimchelly) [reply rest/data/pod.go:30]: Done.

File: rest/route/middleware.go (hunk 2 above)
Comment (Kimchelly) [reply rest/route/middleware.go:325]: Done.

**Review by julianedwards — APPROVED** [review]

**Pull request merged**